src/controller/model/endpoint.ts

```diff
@@ -403,6 +403,68 @@ class Endpoint extends Entity {
         }
     }
 
+    public async writeResponse(
+        clusterKey: number | string, transactionSequenceNumber: number, attributes: KeyValue, options?: Options
+    ): Promise<void> {
+        assert(!options || !options.hasOwnProperty('transactionSequenceNumber'), 'Use parameter');
+        const cluster = Zcl.Utils.getCluster(clusterKey);
+        options = this.getOptionsWithDefaults(options, true, Zcl.Direction.SERVER_TO_CLIENT, cluster.manufacturerCode);
+        const payloadOther: {status: number; attrId: number}[] = [];
+        const payloadSuccess: {status: number}[] = [];
```

Comment:
What if something is in both payloadOther and payloadSuccess? It seems only the last one set by `payload = payloadOther;` is send.

Reply:
According to the ZCL specification, the `status = 0` and the `status != 0` must not be indicated at the same time in the payload, so this situation should never occur, if the user writes the payload correctly.

Wrong `{0x0000: {status: 0x00}, 0x4292: {status: 0x06}, 0xfff1: {status: 0x01}, 0xfff2: {status: 0x01}}`
Correct `{0x4292: {status: 0x06}, 0xfff1: {status: 0x01}, 0xfff2: {status: 0x01}}`
Correct `{0x0000: {status: 0x00}`

I just verified that 'ZclFrame.create' is able to ignore the 'attrId' attribute when the 'status' is 0, so I think it is better to simplify the method as it was before.

Reply:
Changes done.
Sorry, I made the code too complicated